src/windows-cli.cpp

```diff
@@ -0,0 +1,62 @@
+#include <iostream>
+#include <windows.h>
+
+std::wstring joinArgs(int argc, wchar_t* argv[])
+{
+    std::wstring result;
+    for (int i = 1; i < argc; ++i) {
+        if (i > 1) {
+            result += L" ";
+        }
+        result += argv[i];
+    }
+    return result;
+}
+
+void CallFlameshot(const std::wstring args, bool wait)
+{
+    // generate full path for flameshot executable
+    wchar_t path[MAX_PATH];
+    int pathLength = GetModuleFileNameW(NULL, path, MAX_PATH);
+    std::wstring moduleDir = std::wstring(path, pathLength - 18);
+    // generate command string
+    // note: binary path placed within quotes in case of spaces in path
+    int cmdSize = 32 + sizeof(moduleDir) + sizeof(args);
+    wchar_t* cmd = (wchar_t*)malloc(sizeof(wchar_t) * cmdSize);
+    swprintf(cmd,
+             cmdSize,
+             L"\"%s\\flameshot.exe\" %s",
+             moduleDir.c_str(),
+             args.c_str());
+    // call subprocess
+    FILE* stream = _wpopen(cmd, L"r");
+    free(cmd);
+    if (wait) {
+        if (stream) {
+            const int MAX_BUFFER = 2048;
+            char buffer[MAX_BUFFER];
+            while (!feof(stream)) {
+                if (fgets(buffer, MAX_BUFFER, stream) != NULL) {
+                    std::cout << buffer;
+                }
+            }
+        }
+        _pclose(stream);
+    }
+    return;
+}
+
+// Console 'wrapper' for flameshot on windows
+int wmain(int argc, wchar_t* argv[])
+{
+    // if no args, do not wait for stdout
+    if (argc == 1) {
+        std::cout << "Starting flameshot in daemon mode" << std::endl;
+        CallFlameshot(L"", false);
+    } else {
+        std::wstring argString = joinArgs(argc, argv);
+        CallFlameshot(argString, true);
+    }
+    std::cout.flush();
+    return 0;
+}
```

Comment:
I get that Windows isn't as strict as POSIX with its definition of a line or a textfile, but cross toolchains would be technically allowed to toss this file out due to the missing EOL